### ISSUE TITLE
Web - IOU - Delete request pop-up does not close when requestee pays the money request

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#29050]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to staging.new.expensify.com.
2. [User A] Send a money request to User B.
3. [User A] Open the IOU details > click three-dot menu > Delete request > Leave the pop-up open.
4. [User B] Pay the request.

## Expected Result:
The delete pop-up disappears.

## Actual Result:
The delete pop-up does not close although the three-dot menu in the background has disappeared.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** v1.3.79-3

**Reproducible in staging?:** Y

**Reproducible in production?:** Y

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/115492554/18afa238-3dc9-4571-9e90-157a4abb5eb1

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause-Internal Team

**Slack conversation:** @

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by martin12345678900** [comment]
# **Proposal**

## **Please re-state the problem that we are trying to solve in this issue**
When user B approves the request three dots disappear, but delete modal doesn't hide.

## **What is the root cause of the problem**
The root problem is that we are not checking when request is approved in order to close the Delete Request modal.

## **What changes do you think we should make in order to solve the problem?**
We can add an additional useEffect in ```MoneyRequestHeader.js``` component which checks when the request is approved and perform an action to close the modal.

Example code:

```javascript
    useEffect(() => {
        if (isApproved) setIsDeleteModalVisible(false);
    }, [isApproved])
```

## **What alternative solutions did you explore? (Optional)**
N/A

**Comment by abzokhattab** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Delete request confirmation dialog does not close when requestee pays the money request


### What is the root cause of that problem?
we dont sync the `isVisible` of the ConfirmModal with the threedot menu https://github.com/Expensify/App/blob/389d7b0c4b96c6f2a2295055c685791d35b65252/src/components/MoneyRequestHeader.js#L121

### What changes do you think we should make in order to solve the problem?
add 

```js
    const shouldShow = isActionOwner && !isSettled && !isApproved;
```

and refactor the `shouldShowThreeDotsButton` of  `HeaderWithBackButton` to be 

```js
                    shouldShowThreeDotsButton={shouldShow}
```

and finally change the `isVisible` of the `ConfirmModel` to:

```js
                isVisible={isDeleteModalVisible && shouldShow}
```

**Comment by abzokhattab** [comment]
Good point but do we have a case where the three dots appear again? also still the `isDeleteModalVisible` is not used anywhere else other than in this line so shouldn't be a problem.

**Comment by MonilBhavsar** [comment]
Wow, very edge case. 
I am pretty sure even if we don't hide Delete Request when request is settled in parallel, server will return with appropriate response. Happy to fix it because children element is visible and parent is not.
Proposal looks good to me. We need to adjust variable name but can do it in a PR

**Comment by MonilBhavsar** [comment]
Looks like we're good to issue payment here. @Ollyws could you please take a look at the checklist

**Comment by trjExpensify** [comment]
Cool, thanks for that @Ollyws. I agree we don't need a regression test for this. 

I'd be curious if you and @MonilBhavsar have any thoughts on whether implementing an instance of a "double modal" in the future would result in the same bug?.. and if so, is there anything somewhere we can help avoid introducing that bug again?

**Comment by trjExpensify** [comment]
Confirming payments made as follows: 

- $750 to @ZhenjaHorbach for the bug fix
- $750 to @Ollyws for the C+ review

#urgency bonus applied as per [here](https://github.com/Expensify/App/issues/29050#issuecomment-1754384045).

**Comment by trjExpensify** [comment]
We're done here then, closing!
